src/renderer.cpp

```diff
@@ -596,25 +626,50 @@ void Renderer::shutdown()
 
     bgfx::shutdown();
 }
+void Renderer::clearShadowMap()
+{
+    bgfx::setViewClear(0, BGFX_CLEAR_DEPTH, internalState.clearColorValue, 1.0f, 0);
+    bgfx::touch(0);
+}
 
 void Renderer::draw()
 {
     if (trace_all || trace_renderer) trace("Renderer::%s with backendProgram %d\n", __func__, backendProgram);
 
-    // Set current view rect
-    if (backendProgram == RendererProgram::POSTPROCESSING)
-        bgfx::setViewRect(backendViewId, 0, 0, window_size_x, window_size_y);
-    else {
+    auto drawType = getDrawType();
+    if (drawType == DrawType::SHADOW_MAP)

```

Comment:
Ok now I get what you meant with draw types. Please then change the internalState variable and name it generically too! Thanks :)

Reply:
I changed he name to internalState variable name to lightingDrawSettings.

Reply:
As a kind reminder then, consider using the prefix `LIGHT_DRAW_TYPE` in your own const names in the shader code too :) Thanks